### PR TITLE
tenyr 0.9.3 (significant version update)

### PR DESCRIPTION
**Pull request by kulp**
Stop patching (which was incorrect -- see http://bit.ly/2jPhJYX)

Depend on sdl2_image softly

Disable JIT build until we have AsmJit somehow

Add basic test using check_sim

Update install script for new version

Specify that HEAD is on develop branch

Depend on pkg-config during build

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by woodruffw** [comment]
`test` failed, looks like this make target no longer exists:

```
==> make check_sim
make: *** No rule to make target `check_sim'.  Stop.
Error: tenyr: failed
Failed executing: make check_sim
```

**Comment by kulp** [comment]
I think the problem is that I used File.chdir inside the `install` rule. Could that cause this ?

I am new to writing Homebrew recipes. How can I test the `test` part of a recipe locally through `brew` ?

**Comment by woodruffw** [comment]
> I think the problem is that I used File.chdir inside the install rule. Could that cause this ?

Possibly, yeah. You can do this instead:

```ruby
cd "path/to/whatever" do
  # whatever
end
```

> I am new to writing Homebrew recipes. How can I test the test part of a recipe locally through brew ?

`brew test tenyr` should work.

**Comment by kulp** [comment]
I was under the mistaken impression that the `test` method was called within the context of a source build, but I now see why this is not the case. Instead of using a rule from the build makefile, I will create a small inline test case for now.

File: Formula/tenyr.rb [hunk 1]
```diff
@@ -14,43 +14,53 @@ class Tenyr < Formula
...
+    cd builddir do
+      bin.install "tsim", "tas", "tld"
+      lib.install Dir["*.dylib"]
+    end
+  end
 
- struct symbol *symbol_find(struct symbol_list *list, const char *name);
+  test do
+    # sanity test assembler, linker and simulator
+    `echo 'B <- 9'     | tas --output=a.to -`
```
Comment (woodruffw) [inline Formula/tenyr.rb]: These should probably be wrapped with `system` instead, and file creation should be isolated:

```ruby
(testpath/"part1").write "B <- 9"
(testpath/"part2").write "C <- B * 3"

system "tas", "--output=a.to", "part1"
system "tas", "--output=b.to", "part2"
 # etc...

assert_match "C 0000001b", `tsim -vvvv test.texe`
```

Comment (kulp) [inline Formula/tenyr.rb]: Thank you, I will make the suggested changes.

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `tenyr 0.9.3`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Comment by kulp** [comment]
When I had read the contributing guidelines, I wasn't sure where these significant changes fell between the two examples in "The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix`." From now on I think things are clear. Thanks !

**Pull request closed without merging**